ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs

```diff
@@ -87,14 +95,28 @@ defaultPraosTargets =
 -- | These targets are established when Genesis mode is enabled
 -- in node configuration and when the node is syncing up
 --
-defaultGenesisSyncTargets :: PeerSelectionTargets
-defaultGenesisSyncTargets =
-  defaultPraosTargets {
+defaultSyncTargets :: PeerSelectionTargets
+defaultSyncTargets =
+  defaultDeadlineTargets {
     targetNumberOfActivePeers               = 0,
     targetNumberOfKnownBigLedgerPeers       = 100,
     targetNumberOfEstablishedBigLedgerPeers = 50,
     targetNumberOfActiveBigLedgerPeers      = 30 }
 
+-- | This parameter controls the minimum number of active connections
+--   with big ledger peers that must be maintained when syncing in
+--   Genesis mode such that trusted state can be signalled to Consensus.
+--   Exiting syncing / entering deadline mode is predicated on this
+--   condition. This should be below `targetNumberOfActiveBigLedgerPeers`
+--   in `syncTargets` otherwise untrusted state will never be departed.
+--   This value is lower than the target, because in Genesis we may
+--   demote a big ledger peer for underperformance and not immediately
+--   promote one from the warm set in case there are adversaries
+--   whom are intentionally trying to slow us down.
+--
+defaultMinBigLedgerPeersForTrustedState :: MinBigLedgerPeersForTrustedState
+defaultMinBigLedgerPeersForTrustedState = MinBigLedgerPeersForTrustedState 5
```

Comment:
> This should be below `targetNumberOfActiveBigLedgerPeers`
> --   in `syncTargets` otherwise untrusted state will never be departed.


Then we should also check this condition in `sanePeerSelectionTargets`.

Reply:
I would also add a comment in the `targetNumberOfActiveBigLedgerPeers` in `defaulSyncTargets`.

Reply:
The fix is to write tests, it is not broken per se.